testsuite/MDAnalysisTests/core/test_groups.py

```diff
@@ -581,3 +584,137 @@ def test_segmentgroup_repr(self):
+        assert_(np.array_equal(intersect_ab.ix, np.arange(30, 100)))
+        assert_(np.array_equal(a.intersection(b), b.intersection(a)))
+        assert_(len(a.intersection(d)) == 0)
+
+    @staticmethod
+    def _test_difference(a, b, c, d):
+        difference_ab = a.difference(b)
+        assert_(np.array_equal(difference_ab.ix, np.arange(10, 30)))
+
+        difference_ba = b.difference(a)
+        assert_(np.array_equal(difference_ba.ix, np.arange(100, 140)))
+
+        assert_(a.difference(d) == a)
+
+    @staticmethod
+    def _test_symmetric_difference(a, b, c, d):
+        symdiff_ab = a.symmetric_difference(b)
+        assert_(np.array_equal(symdiff_ab.ix,
+                                 np.array(list(range(10, 30))
+                                          + list(range(100, 140)))))
+        assert_(np.array_equal(a.symmetric_difference(b),
+                               b.symmetric_difference(a)))
+
+    @staticmethod
+    def resegment(u, res_per_seg):
+        n_atoms = len(u.atoms)
+        n_residues = len(u.residues)
+        attrs = [
+            attr
+            for attr in u._topology.attrs
+            if not isinstance(attr, Segids)
+        ]
+        limit = int(np.ceil(len(u.residues) / res_per_seg))
+        segids = np.arange(1131).repeat(res_per_seg)[:len(u.residues)]
+        segids = np.array(['s' + str(x) for x in segids], dtype=object)
+        segidx, (segids,) = change_squash((segids,), (segids,))
+        print(segidx)
+        n_segments = len(segids)
+        attrs.append(Segids(segids))
+        top = Topology(n_atoms, n_residues, n_segments,
+                       attrs=attrs,
+                       atom_resindex=u.atoms.resids,
+                       residue_segindex=segidx)
+        return top
+
+    @staticmethod
+    def make_groups(u, level):
+        a = getattr(u, level)[10:100]
+        b = getattr(u, level)[30:140]
+        c = getattr(u, level)[20:50]
+        d = getattr(u, level)[0:0]
+        return a, b, c, d
+
+    def test_groupbase_operators(self):
+        u = mda.Universe(TPR, XTC)
+        # We need our test universe to have at least 140 atoms, residues, and
+        # segments. While it is no issue for atoms and residues, it is for
+        # segments as the universe only has 3. We create a new universe from
+        # the first one, here we set every 10 residues in a new segment.
+        topology = self.resegment(u, res_per_seg=10)
+        u2 = mda.Universe(topology, u.trajectory.filename)
+        for level in ('atoms', 'residues', 'segments'):
+            a, b, c, d = self.make_groups(u2, level)
+            yield self._test_len, a, b, c, d
+            yield self._test_equal, a, b, c, d
+            yield self._test_concatenate, a, b, c, d
+            yield self._test_union, a, b, c, d
+            yield self._test_intersection, a, b, c, d
+            yield self._test_difference, a, b, c, d
+            yield self._test_symmetric_difference, a, b, c, d
```

Comment:
I know you've probably already thought of this, but we also need predictable behaviour when we work with duplicate items in a Group.  Ie `u.atoms[[0, 1, 2, 1, 2]] | u.atoms[[3, 4]] == u.atoms[[0, 1, 2, 3, 4]]`